packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -83,15 +95,19 @@ function Tabs<T extends object>(props: SpectrumTabsProps<T>, ref: DOMRef<HTMLDiv
     if (wrapperRef.current && orientation !== 'vertical') {
       let tabsComponent = wrapperRef.current;
       let tabs = tablistRef.current.querySelectorAll('[role="tab"]');
-      let lastTab = tabs[tabs.length - 1];
+      let tabDimensions = [...tabs].map(tab => tab.getBoundingClientRect());
 
       let end = direction === 'rtl' ? 'left' : 'right';
       let farEdgeTabList = tabsComponent.getBoundingClientRect()[end];
-      let farEdgeLastTab = lastTab?.getBoundingClientRect()[end];
+      let farEdgeLastTab = tabDimensions[tabDimensions.length - 1][end];
       let shouldCollapse = direction === 'rtl' ? farEdgeLastTab < farEdgeTabList : farEdgeTabList < farEdgeLastTab;
       setCollapsed(shouldCollapse);
+      if (tabDimensions.length !== prevTabPositions.current.length || tabDimensions.some((box, index) => box?.left !== prevTabPositions.current[index]?.left && box?.right !== prevTabPositions.current[index]?.right)) {
+        setTabPositions(tabDimensions);
+        prevTabPositions.current = tabDimensions;
```

Comment:
Works well enough, but I wonder if it would instead be sufficient to trigger an arbitrary state change when `children` changes and provide that state to the context in place of `tabLineState`. It would cause a couple of extra `onResize` calls down in TabLine I guess if children changed but the exact sizing of each resulting tab didn't...

Reply:
Right, I thought of doing that, but this is the thing we actually care about changing. And as you point out, some children may change without changing the dimensions, so it saves us some potential extra renders.